### DISCUSSION TITLE
search loader icon doesnt get closed when recieving error from server

### DISCUSSION THREAD

**Discussion by yar-developer** [fomantic/Fomantic-UI#1922] [Behaviors (Api, Form Validation, Visibility)] [answered]
hi, we use the "responseAsync" method of the search element in order to use our own custom API method instead of Fomantic-UI's  method. The search elements response (array of objects) is received in the "results" property. when the API's response state is success(200), everything is fine, because the 'results' property is created and the search's menu gets populated and search's loader turns off automatically.  but when the API response's state is unsuccessful (404 , etc), our web application sends a response with an "error" tag, so the "results" tag doesn't exist in this situation, so here the searches loader keeps on spinning. i want to know that is there a method (behavior) for "search" to manually turn off the loader icon by a Fomantic-UI method?
thank you

**Comment by lubber-de** [comment, marked as answer]
Yes:

```javascript
$('.ui.search').search('remove loading');
```

However, it does nothing more than removing the `loading` class from the `ui search` element, which you can also do manually.
It's not documented, because it was intentionally used internally only. However, each function inside the module can be called as a behavior (for example you can also use `set loading` to enable the loading icon.

**Reply by yar-developer** [reply]
yes of course, i can remove the loading manually, but  i think it's better to remove it using the fomatic-ui framework. in our own project which in some pages we use our own custom loader, when an unsuccess happens, we remove the loader. but fomantic-ui 's 'search' loader only gets removed in success mode. which i think it need's to get developed. what is your opinion?

**Reply by lubber-de** [reply]
As i suggested, you can use the existing behavior "remove loading". Just in case this does something else in a possible future release. ;)

**Reply by yar-developer** [reply]
ok
